src/Product/ProductRepository.php

```diff
@@ -257,11 +282,25 @@ public function find_expiring_product_ids( int $limit = - 1, int $offset = 0 ):
 	 * @return int[] Array of WooCommerce product IDs
 	 */
 	public function find_mc_not_synced_product_ids( int $limit = -1, int $offset = 0 ): array {
-		$types = ProductSyncer::get_supported_product_types();
-		$types = array_diff( $types, [ 'variation' ] );
-		$args  = [
-			'status'     => 'publish',
-			'type'       => $types,
+		// Get syncable products, and swap variation products for its parent.
+		$args = [
+			'status'     => [ 'publish' ],
+			'type'       => array_diff( ProductSyncer::get_supported_product_types(), [ 'variable' ] ),
+			'meta_query' => $this->get_sync_ready_products_meta_query(),
+		];
+
+		$syncable_product_ids = $this->product_helper->maybe_swap_for_parent_ids(
+			array_filter(
+				$this->find_ids( $args, $limit, $offset ),
+				function ( $product_id ) {
+					$product = $this->product_helper->get_wc_product( $product_id );
+					return $this->product_helper->is_sync_ready( $product );
+				}
+			)
+		);
+
+		// Get the product ids which their MC status is 'not synced' from the syncable products.
+		$find_by_ids_args = [
```

Comment:
This part is being refactored by the next commit eb9f841 so it can be ignored. Keeping it in the commit log in case anyone is interested in seeing the different between two approaches.